ui/apps/platform/src/utils/queryService.js

```diff
@@ -42,10 +42,14 @@ function objectToWhereClause(query, delimiter = '+') {
             if (typeof value === 'undefined' || value === '') {
                 return acc;
             }
-            const flatValue = Array.isArray(value) ? value.join() : value;
-            const needsExactMatch =
-                key.toLowerCase().indexOf(' id') !== -1 && value.indexOf(',') === -1;
-            const queryValue = needsExactMatch ? `"${flatValue}"` : flatValue;
+            const valueArray = Array.isArray(value) ? value : [value];
+            const queryValue = valueArray
+                .map((val) => {
+                    const needsExactMatch =
+                        key.toLowerCase().indexOf(' id') !== -1 && val.indexOf(',') === -1;
+                    return needsExactMatch ? `"${val}"` : val;
+                })
+                .join();
```

Comment:
Worth adding a test case for this, but we can do that after the release is cut. OK to merge as-is.